### PR TITLE
Add C++ coding style part to GREP 0001

### PR DESCRIPTION
**Pull request by noc0lour**
Adds a short part about how to submit new code into repo and adhering to the LLVM Coding Style guidelines if doing so.

I'll submit a separate GREP about how I think we can get our repo in a sensible state for automating that. (Learning from other FOSS projects how they did that and how it worked out.)

"A formatting process which is both manual and insufficient is doomed to be abandoned" -- MongoDB developers

### CODE REVIEW

File: grep-0001-coding-guidelines.md [hunk 1]
```diff
@@ -73,6 +72,20 @@ and follow them.
   can thus be optimized better at compile time. In many cases, lambdas are
   also more readable than the `bind` calls, which have add their own
   syntactical idiosyncrasies (such as the `_1` argument etc.).
+* Feel free to use modern C/C++ features even if they were not used before.
+  Make sure they work with the compilers and dependencies which are set for the
+  version of GNU Radio the commit will be made upon. The GNU Radio CI system
+  will be able to confirm this. (Note: C++11 features are available starting
+  with GNU Radio version 3.8).
+
+#### Code Style Guidelines
```
Comment (mbr0wn) [inline grep-0001-coding-guidelines.md]: Can we call formatting rules formatting rules. "Coding Style" is ambiguous, for example, I prefer a functional coding style, which is not what this means.

File: grep-0001-coding-guidelines.md [hunk 2]
```diff
@@ -73,6 +72,20 @@ and follow them.
...
+* Use the [LLVM code formatting guide](https://llvm.org/docs/CodingStandards.html#source-code-formatting). Following points will specify differences to it and highlight important points.
+* Do not indent namespaces
+* Use 4 spaces indentation width for new code.
```
Comment (mbr0wn) [inline grep-0001-coding-guidelines.md]: I like 4 spaces, but none of our code does that. This is a big change, it'll affect future diffs etc.

File: grep-0001-coding-guidelines.md [hunk 3]
```diff
@@ -73,6 +72,20 @@ and follow them.
...
+* Use spaces instead of tabs. If extending an existing file stick with the given indentation level. Do _not_ reindent a whole file and indent code consistently.
+* Before checking in code use tools in the code tree to sanitze your patches
```
Comment (mbr0wn) [inline grep-0001-coding-guidelines.md]: Typo (sanitize). Also, do we want to recommend tools? I think we should.

**Review by mbr0wn — CHANGES_REQUESTED** [review]

File: grep-0001-coding-guidelines.md (hunk 1 above)
Comment (noc0lour) [reply grep-0001-coding-guidelines.md]: :+1: Will do that.

File: grep-0001-coding-guidelines.md (hunk 2 above)
Comment (noc0lour) [reply grep-0001-coding-guidelines.md]: I talked with @marcusmueller about that, and looking through our source code we don't have a clear cut on either side. It's all mixed up, even in single files.

File: grep-0001-coding-guidelines.md (hunk 3 above)
Comment (noc0lour) [reply grep-0001-coding-guidelines.md]: I'm working on a second GREP discussing about how we want to implement changes to formatting going forward. Which will contain how tools should be used (e.g. clang-format, but with a sane wrapper for GNU Radio)

Comment (mbr0wn) [reply grep-0001-coding-guidelines.md]: Looking forward to that! It'll be a great help for you contributors.

Comment (mbr0wn) [reply grep-0001-coding-guidelines.md]: ...and we should probably do a huge formatting patch on both maint-3.7 and master to reset it once and for all.

File: grep-0001-coding-guidelines.md (hunk 2 above)
Comment (mbr0wn) [reply grep-0001-coding-guidelines.md]: Yeah, and at least we'll finally have an actual guideline. Like I said, it's probably worth auto-reformatting the source tree at one point, after that, it'll be much easier to encourage these rules properly.

**Review by marcusmueller — APPROVED** [review]

**Comment by marcusmueller** [comment]
I'm very fine with this change, and I think @mbr0wn's concerns were addressed.

Since I'm the champion of this, I'm setting an arbitrary deadline for further comments/reviews to stop the merging of this: 2018-03-08 12:00UTC.

**Review by mbr0wn — APPROVED** [review]

**Comment by mbr0wn** [comment]
Merged manually 8e96848e6adb3d6a5f00d9f4bf6006605b7b26d0, @marcusmueller's deadline was long overdue.

**Pull request closed without merging**